lbrynet/daemon/Publisher.py

```diff
@@ -54,6 +55,42 @@ def create_and_publish_stream(self, name, bid, claim_dict, file_path, claim_addr
     @defer.inlineCallbacks
     def publish_stream(self, name, bid, claim_dict, claim_address=None, change_address=None):
         """Make a claim without creating a lbry file"""
+        log.info('Starting publish for %s', name)
+
+        source = claim_dict['stream']['source'].get('source', None)
+        source_type = claim_dict['stream']['source'].get('sourceType', None)
+        content_type = claim_dict['stream']['source'].get('contentType', 'application/octet-stream')
+        version = claim_dict['stream']['source'].get('version', '_0_0_1')
+
+        if source is None:
+            raise Exception("Cannot publish a stream with no source")
+
+        if source_type is None:
+            lbry_sd_hash_re = r'^[a-f0-9]{96}$'
+            http_re = r'^https?://[-A-Za-z0-9+&@#/%?=~_|!:,.;]+[-A-Za-z0-9+&@#/%=~_|]'
+            btih_re = r'^[a-f0-9]{40}$'
+
+            lbry_sd_hash = re.findall(lbry_sd_hash_re, source)
+            http = re.findall(http_re, source)
+            btih = re.findall(btih_re, source.lower())
+
+            if len(lbry_sd_hash) == 1:
+                source_type = "lbry_sd_hash"
+            elif len(btih) == 1:
+                source_type = "btih"
+            elif len(http) == 1:
+                source_type = "http"
+            else:
+                raise Exception("Unknown source type")
```

Comment:
As indicated in the above comment, this stuff should go in lbryschema. When you move it there the exception you're looking for is https://github.com/lbryio/lbryschema/blob/60ab79474b4e70e13b8b302f91fdfbcd4772f4ab/lbryschema/error.py#L1